src/repr/src/row.rs

```diff
@@ -898,7 +898,11 @@ impl Row {
 
     /// Returns the total amount of bytes used by this row.
     pub fn byte_len(&self) -> usize {
-        let heap_size = self.data.len().saturating_sub(self.data.inline_size());
+        let heap_size = if self.data.spilled() {
+            self.data.len()
```

Comment:
`size_of_val` was giving a constant value no matter how big or small the actual vec was.